lib/AST/ASTImporter.cpp

```diff
@@ -282,6 +282,34 @@ namespace clang {
         ToD->setImplicit();
     }
 
+    void addDeclToContexts(Decl *FromD, Decl *ToD) {
+      if (!Importer.isMinimalImport()) {
```

Comment:
We could decrease the nesting by starting with `if (Importer.isMinimalImport())` and then do an early return.